service/src/main/java/uk/gov/hmcts/reform/fpl/service/SendDocumentService.java

```diff
@@ -93,12 +95,43 @@ private List<Recipient> getRepresentativesServedByPost(CaseData caseData) {
     }
 
     private List<Recipient> getNotRepresentedRespondents(CaseData caseData) {
-        return unwrapElements(caseData.getRespondents1()).stream()
-            .filter(respondent -> ObjectUtils.isEmpty(respondent.getRepresentedBy())
-                && hasNoLegalRepresentation(respondent))
-            .filter(respondent -> !respondent.isDeceasedOrNFA())
-            .map(Respondent::getParty)
-            .collect(toList());
+        List<Element<Recipient>> partiesWithConfidentialAddress = getPartiesWithConfidentialAddress(caseData);
+
+        if (isNotEmpty(caseData.getRespondents1())) {
+            return caseData.getRespondents1().stream()
+                .filter(respondent -> ObjectUtils.isEmpty(respondent.getValue().getRepresentedBy())
+                    && hasNoLegalRepresentation(respondent.getValue()))
+                .filter(respondent -> !respondent.getValue().isDeceasedOrNFA())
+                .map(respondent ->
+                    ElementUtils.findElement(respondent.getId(), partiesWithConfidentialAddress).isPresent()
```

Comment:
I wonder if we could simplify this a little if we create a helper method getRespondentAddress (and call it here for each respondent) which:

1. Uses what we do in containsConfidentialRespondentDetails to return the confidential address we get from caseData.getConfidentialRespondents() if we can find their uuid in the list of confidential respondents
2. If not just return the address as before.